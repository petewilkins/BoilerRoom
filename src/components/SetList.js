import React, { Component } from 'react';
import { ScrollView } from 'react-native';
import axios from 'axios';
import SetDetail from './SetDetail';

class SetList extends Component {
  state = { sets: [] };

  componentWillMount() {
    axios.get('https://api.myjson.com/bins/5zkr9')
      .then(response => this.setState({ sets: response.data }));
  }

  renderSets() {
    return this.state.sets.map(set =>
      <SetDetail key={set.title} set={set} />
    );
  }

  render() {
    console.log(this.state);

    return (
      <ScrollView>
        {this.renderSets()}
      </ScrollView>
    );
  }
}

export default SetList;
