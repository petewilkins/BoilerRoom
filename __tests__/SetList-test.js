import 'react-native';
import React from 'react';
import SetList from '../src/components/SetList';
import { ScrollView } from 'react-native';
import { shallow } from 'enzyme';

import renderer from 'react-test-renderer';

test('renders the SetList component correctly', () => {
  const tree = renderer.create(
    <SetList />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders one ScrollView component', () => {
  const wrapper = shallow(<SetList />);
  expect(wrapper.find(ScrollView).length).toBe(1);
});
