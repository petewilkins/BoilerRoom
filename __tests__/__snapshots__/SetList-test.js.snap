// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SetDetail component correctly 1`] = `
<RCTScrollView>
  <View />
</RCTScrollView>
`;

exports[`renders the SetList component correctly 1`] = `
<RCTScrollView>
  <View />
</RCTScrollView>
`;
